### ISSUE TITLE
Basic auth support

### ISSUE THREAD

**Issue by ldub** [ipfs-shipyard/py-ipfs-http-client#154]
For people using a private IPFS cluster with a swarm key, it is also useful to have support for HTTP basic auth. Interaction with the gateway is easy, as there is no library and you can just use URLs like `https://user:pass@gatewayurl.com/ipfs/hash`. However, for using the IPFS API, this is not as simple.

Making `py-ipfs-api` support basic auth and is a useful and simple change that I'm willing to implement. Instead of vendoring this code I would love to submit a PR. Just want to ask before doing this - would you accept a PR for basic auth support?

**Comment by brianmcmichael** [comment]
+1 on this. This worked with `ipfsapi` but the new mutliaddr scheme doesn't support it and I'm trying to figure out how to connect.

**Comment by ldub** [comment]
@brianmcmichael could you try https://github.com/ldub/py-ipfs-api/ ? I'm having an issue testing it against my own IPFS node because of some SSL issue, but you should be able to do the following as mentioned in the README:

```
>>> import base64
>>> import ipfshttpclient
>>> creds = base64.b64encode(b"username:password").decode("utf-8")
>>> headers = {"Authorization" : "Basic " + creds}
>>> api = ipfshttpclient.connect('/dns/ipfs-api.example.com/tcp/443/https', headers=headers)
```

**Comment by alexander255** [comment]
The TLS issue should be fixed (see #185), I'm still open to adding support for dedicated `username` and `password` parameters to the codebase btw. :wink:
